profiles/permissions_test.py

```diff
@@ -47,6 +55,135 @@ def test_cant_edit_if_not_owner(self):
         other_user = UserFactory.create()
         for method in ('POST', 'PATCH', 'PUT'):
             with mute_signals(post_save):
-                profile = ProfileFactory.create()
+                profile = ProfileFactory.create(account_privacy=Profile.PUBLIC_TO_MM)
+
             request = Mock(method=method, user=other_user)
             assert not perm.has_object_permission(request, None, profile)
+
+
+# pylint: disable=no-self-use
+class CanSeeIfNotPrivateTests(TestCase):
+    """
+    Tests for CanSeeIfNotPrivate permissions
+    """
+
+    def setUp(self):
+        super(CanSeeIfNotPrivateTests, self).setUp()
+        with mute_signals(post_save):
+            self.other_user = other_user = UserFactory.create()
+            username = "{}_edx".format(other_user.username)
+            social_auth = other_user.social_auth.create(
+                provider=EdxOrgOAuth2.name,
+                uid=username
+            )
+            self.other_user_id = social_auth.uid
+            ProfileFactory.create(user=other_user, verified_micromaster_user=False)
+
+        with mute_signals(post_save):
+            self.profile_user = profile_user = UserFactory.create()
+            username = "{}_edx".format(profile_user.username)
+            social_auth = profile_user.social_auth.create(
+                provider=EdxOrgOAuth2.name,
+                uid=username
+            )
+            self.profile_user_id = social_auth.uid
+
+    def test_cant_view_if_privacy_is_private(self):
+        """
+        Users are not supposed to view private profiles.
+        """
+        perm = CanSeeIfNotPrivate()
+
+        with mute_signals(post_save):
+            ProfileFactory.create(user=self.profile_user, account_privacy=Profile.PRIVATE)
+
+        request = Mock(user=self.other_user)
+        view = Mock(kwargs={'user': self.profile_user_id})
+
+        with self.assertRaises(Http404):
+            perm.has_permission(request, view)
+
+    def test_cant_view_if_anonymous_user(self):
+        """
+        Anonymous are not supposed to view public_to_mm profiles.
+        """
+        perm = CanSeeIfNotPrivate()
+        with mute_signals(post_save):
+            ProfileFactory.create(user=self.profile_user, account_privacy=Profile.PUBLIC_TO_MM)
+
+        request = Mock(user=Mock(return_value=Mock(return_value=True)))
```

Comment:
Note that if you changed `True` to `False` here the test would still pass. This is due to mocks in Python evaluating to true by default. 

The first `return_value` should be `is_anonymous` because the first mock returns a mocked object with `user`, and the second mock should be a mocked object with `is_anonymous`. `return_value` should only be used for when we want to mock a function.